### ISSUE TITLE
RunFlowML does not work when DHF app servers require SSL

### ISSUE THREAD

**Issue by rjrudin** [marklogic/nifi#145]
When RunFlowML creates its connections to the DHF app servers, it is using the `SSLService` and `SSLHostnameVerifier` from the user's SSL service in NiFi, but it is not carrying over the `X509TrustManager`. Because of this, we run into this error:

Caused by: java.lang.RuntimeException: Unable to retrieve flow with name: CurateCustomerJSON; cause: User is not allowed to POST at /data-hub/5/data-services/flow/getFullFlow.sjs
        at com.marklogic.hub.impl.FlowManagerImpl.getFullFlow(FlowManagerImpl.java:156)
        at com.marklogic.hub.flow.impl.FlowRunnerImpl.runFlow(FlowRunnerImpl.java:179)
        at org.apache.nifi.marklogic.processor.RunFlowMarkLogic.onTrigger(RunFlowMarkLogic.java:153)

There's a separate issue here to analyze as to why we get a message like this when the root cause is due to an SSL misconfiguration. I'll follow up on that in the Java Client project with a test case. 

For the NiFi connector, we need to carry over either the user's provided trust store or a "trust anything" trust store.

**Comment by rjrudin** [comment]
Just a note that the keystore/truststore must be configured properly - I did a quick test on a keystore file without the required certificate in it and got the expected error of: "Unable to retrieve flow with name: CurateCustomerJSON; cause: javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target"

Did a quick test where I modified the NiFi connector to use the equivalent of a "simple SSL" SSLContext, and the call succeeded. With the SSLContext from NiFi and a keystore lacking the required certificate, the above failed. With the required certificate in the keystore, the call succeeded.
